jira/resilientsession.py

```diff
@@ -113,11 +113,14 @@ def __recoverable(
                 f"Got ConnectionError [{response}] errno:{response.errno} on {request} {url}\n{vars(response)}\n{response.__dict__}"
             )
         if isinstance(response, Response):
-            if response.status_code in [502, 503, 504, 401]:
-                # 401 UNAUTHORIZED still randomly returned by Atlassian Cloud as of 2017-01-16
+            if response.status_code in [429]:
+                rate_limit_remaining = response.headers["X-RateLimit-Remaining"]
                 msg = f"{response.status_code} {response.reason}"
-                # 2019-07-25: Disabled recovery for codes above^
-                return False
+                logging.warning(
+                    f"""Request rate limited by Jira: number of tokens remaining {rate_limit_remaining}. Consider adding exemption for the user as explained in
+                    https://confluence.atlassian.com/adminjiraserver/improving-instance-stability-with-rate-limiting-983794911.html
+                    """
```

Comment:
```suggestion
                    f"Request rate limited by Jira: number of tokens remaining {rate_limit_remaining}.\n" 
                    + "Consider adding an exemption for the user as explained in: "
                    + "https://confluence.atlassian.com/adminjiraserver/improving-instance-stability-with-rate-limiting-983794911.html"
                    """
```

Technically we could have implicit string concat without the `+`, but I think this may make `black` behave better.

#### Why?


`"""` in this way will actually print all the indents!

https://www.online-python.com/
```python
if True:
    if True:
        if True:
            print(f"""Request rate limited by Jira: number of tokens remaining {5}. Consider adding exemption for the user as explained in
                  https://confluence.atlassian.com/adminjiraserver/improving-instance-stability-with-rate-limiting-983794911.html
                  """)
```

Gives:
```
Request rate limited by Jira: number of tokens remaining 5. Consider adding exemption for the user as explained in
                  https://confluence.atlassian.com/adminjiraserver/improving-instance-stability-with-rate-limiting-983794911.html
                  
```

Reply:
Cool I didn't know about that, thanks for the detailed info! Python is not my main language, I can put together some working code but I'm lacking knowledge on these details :sweat_smile:

Reply:
No problem at all! I think you have an excellent grasp already